component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/token2/Token2Authenticator.java

```diff
@@ -0,0 +1,286 @@
+                while (inputLine != null) {
+                    builder.append(inputLine).append("\n");
+                    inputLine = in.readLine();
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("response: " + builder.toString());
+                }
+            } catch (MalformedURLException e) {
+                throw new AuthenticationFailedException("Invalid URL", e);
+            } catch (ProtocolException e) {
+                throw new AuthenticationFailedException("Error while setting the HTTP request method", e);
+            } catch (IOException e) {
+                throw new AuthenticationFailedException("Error in I/O Streams" + e.getMessage(), e);
+            } finally {
+                try {
+                    if (in != null) {
+                        in.close();
+                    }
+                } catch (IOException e) {
+                    log.error("Couldn't close the I/O Streams", e);
+                }
+                if (urlConnection != null) {
+                    urlConnection.disconnect();
+                }
+            }
+            return builder.toString();
+        }
+    }
+
+    /**
+     * Get the user realm of the logged in user
+     */
+    private UserRealm getUserRealm(String username) throws AuthenticationFailedException {
+        UserRealm userRealm;
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
+            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+            RealmService realmService = IdentityTenantUtil.getRealmService();
+            userRealm = (UserRealm) realmService.getTenantUserRealm(tenantId);
+        } catch (Exception e) {
+            throw new AuthenticationFailedException("Cannot find the user realm", e);
+        }
+        return userRealm;
+    }
+
+    /**
+     * Get the username of the logged in User
+     */
+    private String getUsername(AuthenticationContext context) {
+        String username = null;
+        for (Integer stepMap : context.getSequenceConfig().getStepMap().keySet())
+            if (context.getSequenceConfig().getStepMap().get(stepMap).getAuthenticatedUser() != null
+                    && context.getSequenceConfig().getStepMap().get(stepMap).getAuthenticatedAutenticator()
+                    .getApplicationAuthenticator() instanceof LocalApplicationAuthenticator) {
+                username = String.valueOf(context.getSequenceConfig().getStepMap().get(stepMap).getAuthenticatedUser());
+                break;
+            }
+        return username;
+    }
+
+    /**
+     * Get the configuration properties of UI
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+        List<Property> configProperties = new ArrayList<>();
+
+        Property apiKey = new Property();
+        apiKey.setName(Token2Constants.APIKEY);
+        apiKey.setDisplayName("Api Key");
```

Comment:
fixed in https://github.com/wso2-extensions/identity-outbound-auth-token2/pull/2